spark-java/src/main/java/com/baeldung/sparkjava/UserService.java

```diff
@@ -0,0 +1,13 @@
+package com.baeldung.sparkjava;
+
+import java.util.Collection;
+import java.util.HashMap;
+
+public interface UserService {
+    public void addUser (User user) ;
+    public Collection<User> getUsers () ;
+    public User getUser (String id) ;
+    public User editUser (String id, HashMap userArg) throws Exception;
```

Comment:
1. Change HashMap to Map (program to the interface, not to the implementation).
Or consider that it is more common in a REST service for the PUT operation to accept an entire JSON object instead of only the fields to be changed like you do here using the Map, and instead you can just implement it as public void User editUser(User user) ...

2. Don't declare that something throws 'Exception'. Instead, create your own exception class (e.g. UserException) and throw that.